Remove Git dep fixes #14

bil-elmoussaoui opened the pull request.

Foggalong commented on gen.py:
```diff
@@ -53,6 +52,12 @@ def mkdir(dir):
 
+def clone(repo):
+    p = Popen(["git", "clone", repo], stdout=PIPE, stderr=PIPE)
+    output, error = p.communicate()
```
What does the output vairable do here?

Foggalong commented on gen.py:
```diff
@@ -53,6 +52,12 @@ def mkdir(dir):
+    p = Popen(["git", "clone", repo], stdout=PIPE, stderr=PIPE)
+    output, error = p.communicate()
+    if error:
+        exit("Please install 'git' first. Exiting.")       
```
Would I be correct in thinking this error will also display if any other error is received? (e.g. the user isn't connected to the internet, the repo folder already exists, etc) That's why the exception is handled as is in lines 91 & 92

bil-elmoussaoui replied:
The result of p.communicate is a tuple (output and error). As we don't wait
any result we will catch only errors.
And i can add a verification if the directory already exists. The same for
the internet connexion. I will try to do that tonight.

Le 14 août 2016 00:38, "Joshua H. Fogg" <notifications@github.com> a écrit :

> In gen.py
> <https://github.com/numixproject/numix-core/pull/18#discussion_r74693085>:
>
> > @@ -53,6 +52,12 @@ def mkdir(dir):
> >      if not path.exists(dir):
> >          makedirs(dir)
> >
> > +def clone(repo):
> > +    p = Popen(["git", "clone", repo], stdout=PIPE, stderr=PIPE)
> > +    output, error = p.communicate()
> > +    if error:
> > +        exit("Please install 'git' first. Exiting.")
>
> Would I be correct in thinking this error will also display if any other
> error is received? (e.g. the user isn't connected to the internet, the repo
> folder already exists, etc) That's why the exception is handled as is in
> lines 91 & 92
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/numixproject/numix-core/pull/18/files/8a546174d125a4e6cb582295cb618921e6d403aa#r74693085>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AHTlxWRKZIlc_LrUR4_FRMoX8U3WSSzaks5qfkdTgaJpZM4JhUXq>
> .
>

Foggalong replied:
The point about errors was more that there are more errors that could occur than is sensible to cover with cases. A general solution which just displays the error and exits would be more suitable.

bil-elmoussaoui replied:
Well i can do that ! :p

Le 14 août 2016 13:06, "Joshua H. Fogg" <notifications@github.com> a écrit :

> In gen.py
> <https://github.com/numixproject/numix-core/pull/18#discussion_r74699914>:
>
> > @@ -53,6 +52,12 @@ def mkdir(dir):
> >      if not path.exists(dir):
> >          makedirs(dir)
> >
> > +def clone(repo):
> > +    p = Popen(["git", "clone", repo], stdout=PIPE, stderr=PIPE)
> > +    output, error = p.communicate()
> > +    if error:
> > +        exit("Please install 'git' first. Exiting.")
>
> The point about errors was more that there are more errors that could
> occur than is sensible to cover with cases. A general solution which just
> displays the error and exits would be more suitable.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/numixproject/numix-core/pull/18/files/8a546174d125a4e6cb582295cb618921e6d403aa#r74699914>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AHTlxQ17kpldvnrRFDjhbg_JWmL-RcdCks5qfvalgaJpZM4JhUXq>
> .
>

bil-elmoussaoui commented:
Done!

Foggalong commented:
Excellent, just needs an update to the README now to reflect the changes now :+1:

bil-elmoussaoui commented:
I will do in few seconds

Le 19 août 2016 13:25, "Joshua H. Fogg" <notifications@github.com> a écrit :

> Excellent, just needs an update to the README now to reflect the changes
> now 👍
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/numixproject/numix-core/pull/18#issuecomment-240993979>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AHTlxfu4A4UDPC3FYMDpJIcFVp2k1t-Zks5qhZKxgaJpZM4JhUXq>
> .
>

bil-elmoussaoui commented:
Done :+1:

Foggalong commented:
Just checked up on this and it isn't working for me, the message stating the clone repository is tripping the error switch.
![screenshot_2016-08-20_00-21-19](https://cloud.githubusercontent.com/assets/5920259/17826782/26bd26cc-666c-11e6-9436-131584970350.png)

bil-elmoussaoui commented:
I can't get this done without having a Linux machine to test on it :( Can you help me out debuging this by adding a  print(error) just after  output, error = p.communicate()

bil-elmoussaoui commented:
Fixed ! I added few stuff to make sure everyting works fine in almost every case
- Check if git installed
- Check if there's an internet connection 
- Check if `data.json` exists
- Also fixes #22 see the screenshot below
![capture du 2016-09-18 14-41-15](https://cloud.githubusercontent.com/assets/7660997/18615723/0afb9f0a-7dae-11e6-865f-278e1a23297b.png)

Foggalong commented:
Tested the latest and it's still broken for me :frowning: 
![screenshot_2016-10-06_21-12-07](https://cloud.githubusercontent.com/assets/5920259/19168771/a5287026-8c09-11e6-8755-03cb37db6c8a.png)
